components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/throttling/ThrottleHandler.java

```diff
@@ -229,127 +230,131 @@ private boolean doRoleBasedAccessThrottlingWithCEP(MessageContext synCtx, Config
+                                    List<ConditionGroupDTO> applicableConditions = getThrottleConditionEvaluator()
+                                            .getApplicableConditions(synCtx, authContext, conditionGroupDTOs);
+                                    for (ConditionGroupDTO conditionGroup : applicableConditions) {
+                                        String combinedResourceLevelThrottleKey = resourceLevelThrottleKey +
+                                                conditionGroup.getConditionGroupId();
+                                        if (log.isDebugEnabled()) {
+                                            log.debug("Checking condition : " + combinedResourceLevelThrottleKey);
+                                        }
 
-                                    if (getThrottleDataHolder().
-                                            isThrottled(combinedResourceLevelThrottleKey)) {
-                                        if (!apiLevelThrottledTriggered) {
-                                            isResourceLevelThrottled = isThrottled = true;
-                                        } else {
-                                            isApiLevelThrottled = isThrottled = true;
+                                        if (getThrottleDataHolder().
+                                                isThrottled(combinedResourceLevelThrottleKey)) {
+                                            if (!apiLevelThrottledTriggered) {
+                                                isResourceLevelThrottled = isThrottled = true;
+                                            } else {
+                                                isApiLevelThrottled = isThrottled = true;
+                                            }
+                                            long timestamp = getThrottleDataHolder().
+                                                    getThrottleNextAccessTimestamp(combinedResourceLevelThrottleKey);
+                                            synCtx.setProperty(APIThrottleConstants.THROTTLED_NEXT_ACCESS_TIMESTAMP,
+                                                    timestamp);
+                                            break;
                                         }
-                                        long timestamp = getThrottleDataHolder().
-                                                getThrottleNextAccessTimestamp(combinedResourceLevelThrottleKey);
-                                        synCtx.setProperty(APIThrottleConstants.THROTTLED_NEXT_ACCESS_TIMESTAMP, timestamp);
-                                        break;
                                     }
                                 }
+
                             }
 
+                        } else {
+                            log.warn("Unable to find throttling information for resource and http verb. Throttling "
+                                    + "will not apply");
```

Comment:
Shouldn't the grammar needs to be corrected as "Throttling will not be applied".

Reply:
Fixed with 2a6fbdd3ac9ce54ca7bc94a45328a41826836a80